### PR TITLE
Feature/global manager

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

**Review by KingYes — DISMISSED** [review]

**Review by KingYes — APPROVED** [review]

File: assets/dev/js/editor/editor-base.js [hunk 1]
```diff
@@ -1067,7 +1068,7 @@ export default class EditorBase extends Marionette.Application {
 
 	addWidgetsCache( widgets ) {
 		jQuery.each( widgets, ( widgetName, widgetConfig ) => {
-			this.widgetsCache[ widgetName ] = jQuery.extend( {}, this.widgetsCache[ widgetName ], widgetConfig );
+			this.widgetsCache[ widgetName ] = jQuery.extend( true, {}, this.widgetsCache[ widgetName ], widgetConfig );
```
Comment (kobizz) [inline assets/dev/js/editor/editor-base.js]: It's potentially very dangerous. You should make sure it'snot breaking anything

File: assets/dev/scss/editor/panel/_elements.scss [hunk 2]
```diff
@@ -25,20 +18,20 @@
...
-	align-items: center;
-	justify-content: center;
-	background-color: rgba(255, 255, 255, 0.5);
-	z-index: 1;
+#elementor-panel-state-loading {
+	display: none;
 
-	body.elementor-controls-ready & {
-		display: none;
+	body.elementor-panel-loading & {
```
Comment (kobizz) [inline assets/dev/scss/editor/panel/_elements.scss]: What happened here?

File: assets/dev/scss/frontend/_global.scss [hunk 3]
```diff
@@ -26,9 +26,6 @@
 	img {
 		height: auto;
 		max-width: 100%;
-		border: none;
-		border-radius: 0;
-		box-shadow: none;
```
Comment (kobizz) [inline assets/dev/scss/frontend/_global.scss]: And here?

File: core/common/assets/js/api/core/shortcuts.js [hunk 4]
```diff
@@ -120,10 +130,19 @@ export default class Shortcuts {
 		}
 
 		const namespace = component.getNamespace(),
-			rootScope = component.getRootContainer();
+			namespaceRoot = component.getRootContainer();
+
+		const filteredByNamespace = scopes.some( ( scope ) => {
+			return namespace === scope;
+		} );
```
Comment (kobizz) [inline core/common/assets/js/api/core/shortcuts.js]: You can shorten it: `const filteredByNamespace = scopes.some( ( scope ) => namespace === scope );`

File: core/common/assets/js/api/core/shortcuts.js [hunk 5]
```diff
@@ -120,10 +130,19 @@ export default class Shortcuts {
...
+
+		if ( filteredByNamespace ) {
+			return true;
+		}
 
+		// Else filter by namespaceRoot.
 		return scopes.some( ( scope ) => {
-			return namespace === scope || rootScope === scope;
+			return namespaceRoot === scope;
 		} );
```
Comment (kobizz) [inline core/common/assets/js/api/core/shortcuts.js]: See above

File: core/common/assets/js/utils/helpers.js [hunk 6]
```diff
@@ -12,13 +12,20 @@ export default class Helpers {
...
 
-		console.warn( message, style, '' ); // eslint-disable-line no-console
+		this.consoleWarn( message );
+	}
+
+	consoleWarn( ...args ) {
+		const style = `font-size: 12px; background-image: url("${ elementorCommon.config.urls.assets }images/logo-icon.png"); background-repeat: no-repeat; background-size: contain;`;
+
+		args.push( style );
+		args.push( '' );
```
Comment (kobizz) [inline core/common/assets/js/utils/helpers.js]: `args.push( style, '' );`

File: core/kits/assets/css/panel.scss [hunk 7]
```diff
@@ -1,3 +1,5 @@
+$editor-kit-accent: #4ab7f4;
```
Comment (kobizz) [inline core/kits/assets/css/panel.scss]: It should be moved to the editor's variables file

File: core/kits/assets/css/panel.scss [hunk 8]
```diff
@@ -10,7 +12,7 @@
 .elementor-editor-kit {
```
Comment (kobizz) [inline core/kits/assets/css/panel.scss]: It's highly recommended to move the whole "kit" style rules to a separated file

File: assets/dev/scss/editor/panel/_elements.scss (hunk 2 above)
Comment (matipojo) [reply assets/dev/scss/editor/panel/_elements.scss]: The "Loading" state is now not only for controls. it works according to the body class `elementor-panel-loading`

File: assets/dev/scss/frontend/_global.scss (hunk 3 above)
Comment (matipojo) [reply assets/dev/scss/frontend/_global.scss]: These rules are now controlled by the Theme Style editor.

File: core/kits/assets/css/panel.scss (hunk 8 above)
Comment (matipojo) [reply core/kits/assets/css/panel.scss]: ?

File: core/kits/assets/css/panel.scss (hunk 7 above)
Comment (matipojo) [reply core/kits/assets/css/panel.scss]: It's a private var

**Pull request merged**